### PR TITLE
Upgrade Prettier to 1.2.2, enable --jsx-bracket-same-line=true

### PR DESCRIPTION
**Pull request by benvinegar**
After this merges, everyone needs to:

* `yarn install` to upgrade Prettier to 1.2.2
* If you're using Prettier in your text editor, enable the `jsx-bracket-same-line` Prettier option.

Note I added a little check to the `pre-commit` hook that verifies Prettier is up to date before running it. This could use some work, and maybe just yarn install on your behalf. Thoughts? (cc @mattrobenolt @mitsuhiko) My goal is to avoid a situation where somebody regresses style on an entire file because they're running an old version of Prettier.

cc @getsentry/javascript

### CODE REVIEW

File: config/hooks/pre-commit [hunk 1]
```diff
@@ -19,6 +19,8 @@ if 'VIRTUAL_ENV' in os.environ:
     sys.path.insert(0, site_packages)
 
 
+PRETTIER_VERSION = "1.2.2"
```
Comment (benvinegar) [inline config/hooks/pre-commit]: Should I try to read this from `package.json`?

Comment (mattrobenolt) [inline config/hooks/pre-commit]: That'd probably be more reliable, no? Assuming that you 100% of the time pin explicitly and don't do like, `^2` or other non-explicit identifiers.

File: config/hooks/pre-commit [hunk 2]
```diff
@@ -32,12 +34,22 @@ def js_format(file_list=None):
         echo('!! Skipping JavaScript formatting because prettier is not installed.')
         return False
 
+    proc = subprocess.Popen([prettier_path, '--version'], stdout=subprocess.PIPE)
```
Comment (mattrobenolt) [inline config/hooks/pre-commit]: In the future, might make more sense to simply use `subprocess.check_output` https://docs.python.org/2/library/subprocess.html#subprocess.check_output Which just encompasses this gross logic. It's python 2.7 specific, but we require that. :)

**Review by mattrobenolt — APPROVED** [review]

File: config/hooks/pre-commit [hunk 3]
```diff
@@ -28,16 +31,33 @@ def js_format(file_list=None):
...
 
     if not os.path.exists(prettier_path):
-        from click import echo
-        echo('!! Skipping JavaScript formatting because prettier is not installed.')
+        echo('!! Skipping JavaScript formatting because prettier is not installed.', err=True)
+        return False
+
+    # Get Prettier version from package.json
+    package_version = None
+    package_json_path = os.path.join((project_root), 'package.json')
```
Comment (mattrobenolt) [inline config/hooks/pre-commit]: Not sure why `project_root` is in parens.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**